static/js/components/SpatialCatalogTable.jsx

```diff
@@ -0,0 +1,195 @@
+      palette: theme.palette,
+      overrides: {
+        MUIDataTablePagination: {
+          toolbar: {
+            flexFlow: "row wrap",
+            justifyContent: "flex-end",
+            padding: "0.5rem 1rem 0",
+            [theme.breakpoints.up("sm")]: {
+              // Cancel out small screen styling and replace
+              padding: "0px",
+              paddingRight: "2px",
+              flexFlow: "row nowrap",
+            },
+          },
+          tableCellContainer: {
+            padding: "1rem",
+          },
+          selectRoot: {
+            marginRight: "0.5rem",
+            [theme.breakpoints.up("sm")]: {
+              marginLeft: "0",
+              marginRight: "2rem",
+            },
+          },
+        },
+      },
+    })
+  );
+
+const SpatialCatalogTable = ({
+  catalog,
+  totalMatches,
+  setSelectedSpatialCatalogEntryId,
+  pageNumber = 1,
+  numPerPage = 10,
+  serverSide = false,
+  hideTitle = false,
+}) => {
+  const classes = useStyles();
+  const theme = useTheme();
+
+  if (!catalog || catalog.entries.length === 0) {
+    return <p>No entries available...</p>;
+  }
+
+  const renderData = (dataIndex) => {
+    const entry = catalog.entries[dataIndex];
+    return <div>{JSON.stringify(entry.data)}</div>;
+  };
+
+  const renderRetrieveSources = (dataIndex) => {
+    const entry = catalog.entries[dataIndex];
+    if (!entry.entry_name) {
+      return <div />;
+    }
+    return (
+      <div>
+        <Button
+          primary
+          onClick={() => {
+            setSelectedSpatialCatalogEntryId(entry.id);
+          }}
+          size="small"
+          type="submit"
+          data-testid={`retrieveSources_${entry.id}`}
+        >
+          Retrieve Sources
+        </Button>
+      </div>
+    );
+  };
+
+  const columns = [
+    {
+      name: "entry_name",
+      label: "Entry Name",
+    },
+    {
+      name: "data",
+      label: "Entry data",
+      options: {
+        customBodyRenderLite: renderData,
+      },
+    },
+    {
+      name: "retrieve_sources",
+      label: "Retrieve Sources",
+      options: {
+        filter: false,
+        sort: true,
+        sortThirdClickReset: true,
+        customBodyRenderLite: renderRetrieveSources,
+      },
+    },
+  ];
+
+  const options = {
+    search: true,
+    selectableRows: "none",
+    elevation: 0,
+    page: pageNumber - 1,
+    rowsPerPage: numPerPage,
+    rowsPerPageOptions: [2, 10, 25, 50, 100],
+    jumpToPage: true,
+    serverSide,
+    pagination: true,
+    count: totalMatches,
+  };
+
+  return (
+    <div>
+      {catalog.entries ? (
+        <Paper className={classes.container}>
+          <StyledEngineProvider injectFirst>
+            <ThemeProvider theme={getMuiTheme(theme)}>
+              <MUIDataTable
+                title={!hideTitle ? "Catalog Entries" : ""}
+                data={catalog.entries}
+                options={options}
+                columns={columns}
+              />
```

Comment:
There is an issue with the `download` option. Basically the `entry_data` column just shows `[Object object]` because its a dict inside, and by default MUI tables don't handle that. So it would need a proper downloadCallback to get it right. If its not important yet and won't be fixed in this PR, please consider deactivating the download button of the table to avoid confusion.

Reply:
Yeah for now I am turning off download on those other columns. These JSON payload ones don't work well in the CSV stuff.